build/periodic.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/sh
+
+# Validate the pipeline is up to date and that no failed prow jobs exist
+
+COMPONENT_ORG=open-cluster-management
+CHECK_RELEASES="2.3 2.4 2.5"
+
+# Clone the repositories needed for this script to work
+cloneRepos() {
+	git clone https://github.com/$COMPONENT_ORG/policy-grc-squad.git
+	git clone https://github.com/$COMPONENT_ORG/pipeline.git
+	mypath=$(pwd)
+	REPOS=$(cat policy-grc-squad/main-branch-sync/repo.txt | grep policy | grep -v framework)
+	for repo in $REPOS; do
+		printf '%s\n' "Updating $repo ...."
+		git clone git@github.com:$repo.git $repo
+	done
+}
+
+# return the most recent git sha for a repository's release branch
+getGitSha() {
+	component=$1
+	release=release-$2
+	cd $COMPONENT_ORG/$component
+	co=`git checkout $release`
+	GITSHA=`git log -n 1 --no-decorate --pretty=oneline | awk '{print $1}'`
+	echo "$GITSHA"
+	cd $BASEDIR
+}
+
+getPipelineValue() {
+	component="$1"
+	release="${2}-integration"
+	key="$3"
+
+	cd pipeline
+	co=$(git checkout "$release")
+	value=`jq '.[] |select(.["image-name"] == "'$component'") | .["'$key'"]' manifest.json | sed 's/"//g'`
+	echo "$value"
+	cd $BASEDIR
+}
+
+checkProwJob() {
+	component="$1"
+	release="$2"
+
+	rc=0
+	# This is a hack to get data from the openshift ci prow
+	# sample curl: curl -H "content-type: application/xml" https://prow.ci.openshift.org/job-history/gs/origin-ci-test/logs/branch-ci-open-cluster-management-iam-policy-controller-release-2.5-publish
+        # which contains this:   var allBuilds = [{"SpyglassLink":"/view/gs/origin-ci-test/logs/branch-ci-open-cluster-management-iam-policy-controller-release-2.5-publish/1468625788839399424","ID":"1468625788839399424","Started":"2021-12-08T16:57:31Z","Duration":103000000000,"Result":"FAILURE","Refs":{"org":"open-cluster-management","repo":"iam-policy-controller","repo_link":"https://github.com/open-cluster-management/iam-policy-controller","base_ref":"release-2.5","base_sha":"567b3597e8324a4c56ec8f1d717ae15d9671e4a8","base_link":"https://github.com/open-cluster-management/iam-policy-controller/compare/d544db4214b4...567b3597e832"}}];
+	echo "Checking prow jobs for a failure with component $component."
+	OUTPUT=$(curl -s https://prow.ci.openshift.org/job-history/gs/origin-ci-test/logs/branch-ci-${COMPONENT_ORG}-${component}-release-${release}-publish)
+	JSON=$(echo "$OUTPUT" | grep "var allBuilds" | sed 's/  var allBuilds =//' | sed 's/;$//' | jq '.[0]')
```

Comment:
Holy moly--great find on the `allBuilds` variable!

And I see this only checks the most recent job. I was thinking of getting statuses for everything but I like this approach better since if one fails but the next one succeeds, that should correct any error.